src/blueprint/definitions.rs

```diff
@@ -525,13 +519,7 @@ pub fn to_definitions<'a>() -> TryFold<'a, ConfigModule, Vec<Definition>, String
 
         Valid::from_iter(config_module.types.iter(), |(name, type_)| {
             let dbl_usage = input_types.contains(name) && output_types.contains(name);
-            if let Some(variants) = &type_.variants {
-                if !variants.is_empty() {
-                    to_enum_type_definition(name, type_, variants).trace(name)
-                } else {
-                    Valid::fail("No variants found for enum".to_string())
```

Comment:
I think this check is still relevant. Can we add it back and also add a test for it.